internal/gatewayapi/translator.go

```diff
@@ -546,11 +548,129 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
+							if redirect := filter.RequestRedirect; redirect != nil {
+								if redirect.Scheme != nil {
+									// Note that gateway API may support additional schemes in the future, but unknown values
+									// must result in an UnsupportedValue status
+									if *redirect.Scheme == "http" || *redirect.Scheme == "https" {
+										irRedirect.Scheme = redirect.Scheme
+									} else {
+										errMsg := fmt.Sprintf("Scheme: %s is unsupported, only 'https' and 'http' are supported", *redirect.Scheme)
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											errMsg,
+										)
+										continue
+									}
+								}
+
+								if redirect.Hostname != nil {
+									if valid, reason := isValidHostname(string(*redirect.Hostname)); !valid {
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											reason,
+										)
+										continue
+									} else {
+										redirectHost := string(*redirect.Hostname)
+										irRedirect.Hostname = &redirectHost
+									}
+								}
+
+								if redirect.Path != nil {
+									switch redirect.Path.Type {
+									case "ReplaceFullPath":
+										if redirect.Path.ReplaceFullPath != nil {
+											irRedirect.Path = &ir.HTTPPathModifier{
+												FullReplace: redirect.Path.ReplaceFullPath,
+											}
+										}
+									case "ReplacePrefixMatch":
+										if redirect.Path.ReplacePrefixMatch != nil {
+											irRedirect.Path = &ir.HTTPPathModifier{
+												PrefixMatchReplace: redirect.Path.ReplacePrefixMatch,
+											}
+										}
+									default:
+										errMsg := fmt.Sprintf("Redirect path type: %s is invalid, only \"ReplaceFullPath\" and \"ReplacePrefixMatch\" are supported", redirect.Path.Type)
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											errMsg,
+										)
+										continue
+									}
+								}
+
+								if redirect.StatusCode != nil {
+									redirectCode := int32(*redirect.StatusCode)
+									// Envoy supports 302, 303, 307, and 308, but gateway API only includes 301 and 302
+									if redirectCode == 301 || redirectCode == 302 {
+										irRedirect.StatusCode = &redirectCode
+									} else {
+										errMsg := fmt.Sprintf("Status code %d is invalid, only 302 and 301 are supported", redirectCode)
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											errMsg,
+										)
+										continue
+									}
+								}
+
+								if redirect.Port != nil {
+									redirectPort := uint32(*redirect.Port)
+									irRedirect.Port = &redirectPort
```

Comment:
We're ending up with a lot of silly code that looks like this. Perhaps we should add a utility function

```go
func ptrTo[T any](x T) *T {
    return &x
}
```

Reply:
Very true, but I kind of feel like this should be a separate PR after this just to keep the logical size smaller.